internal/fs/fs.go

```diff
@@ -926,7 +947,18 @@ func (fs *fileSystem) lookUpOrCreateChildInode(
 	getLookupResult := func() (*inode.Core, error) {
 		parent.Lock()
 		defer parent.Unlock()
-		return parent.LookUpChild(ctx, childName)
+
+		core, err := parent.LookUpChild(ctx, childName)
+		if err == nil && parent.IsBaseDirInode() {
```

Comment:
@sethiay as discussed offline, this code is specific to multi-bucket mount. This is creating (if not already created) and inserting a type-cache-bucket-view for this bucket (called childName) into the (bucketName -> type-cache-bucket-view') map in fs. This is the code which comes up when a new bucket is mounted in dynamic-mount.

Reply:
done